scalableminds/webknossos
# Fix Shift + Scroll for some mouses on macOS

For mouses that only have a vertical scroll wheel, MacOS seems to emulate horizontal scrolling when using `Shift` and vertical scrolling. See for example, this issue: https://bugs.mojang.com/browse/MC-121772?page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel&showAll=true
I fixed it by picking the absolutely larger value of `deltaX` or `deltaY`. As we're not making a difference between horizontal/vertical scrolling afaik, this shouldn't negatively affect other parts of the code.

~~During testing, I noticed that neither `Shift + Scroll` nor `Alt + Scroll` seems to work in Firefox. Not sure whether that is a known bug.~~ This problem was caused by the `imwheel` program, not related to webKnossos.

### URL of deployed dev instance (used for testing):
- https://fixmacscroll.webknossos.xyz

### Steps to test:
- Own a Mac
  - Open a volume tracing and activate the "Brush".
  - Use `Shift + Scroll` to change the size of the brush. This should work in both directions (increasing/decreasing size).
- Test that other mouse bindings using the scroll wheel still work:
  - `Scroll` to move in z-direction
  - `Alt + Scroll` to zoom

### Issues:
- fixes #4854 

------
